### ISSUE TITLE
Have a proper way to customize the "not found" handler for each method

### ISSUE THREAD

**Issue by LoicPoullain** [FoalTS/foal#750]
# Issue

There is currently no proper way to override the *not found* found handler for each method.

## Workaround 1

```typescript
export AppController {
  @Get('*')
  handleGetNotFound() { ... }
  @Post('*')
  handlePostNotFound() { ... }
  @Patch('*')
  handlePatchNotFound() { ... }
  // etc
}
```

## Workaround 2

```typescript
const app = createApp(AppController, {
  postMiddlewares: [
    (req, res, next) => res.json({message: 'my custom 404'})
  ]
});
```

**Comment by LoicPoullain** [comment]
# Solution 1 (proposal)

Add a `@All` method decorator.

```typescript
export AppController {
  @All('*')
  handleError() { ... }
}
```

**Comment by LoicPoullain** [comment]
# Solution 2 (proposal)

```typescript
  const app = createApp(AppController, {
    methods: {
      handleNotFound: true
    }
  });
```

```typescript
export AppController {
  handleNotFound() { ... }
}
```

**Comment by tomer-yechiel** [comment]
i think postMiddlewares is a good and generic solution instead of adding a new configuration for that specific requirement.

**Comment by LoicPoullain** [comment]
The `@All` decorator will be added in next release (solution 1 is chosen).

**Comment by LoicPoullain** [comment]
If someone wished to submit a PR, the file to update is located in `http-responses.ts`

**Comment by LoicPoullain** [comment]
Resolved in v2.1
